src/main/java/com/stripe/net/ApiResource.java

```diff
@@ -153,16 +129,22 @@ private static String subresourceUrl(Class<?> clazz, String id, Class<?> subClaz
   }
 
   /** URL-encodes a string. */
-  public static String urlEncode(String str) throws UnsupportedEncodingException {
+  public static String urlEncode(String str) {
     // Preserve original behavior that passing null for an object id will lead
     // to us actually making a request to /v1/foo/null
     if (str == null) {
       return null;
-    } else {
+    }
+
+    try {
       // Don't use strict form encoding by changing the square bracket control
       // characters back to their literals. This is fine by the server, and
       // makes these parameter strings easier to read.
       return URLEncoder.encode(str, CHARSET).replaceAll("%5B", "[").replaceAll("%5D", "]");
+    } catch (UnsupportedEncodingException e) {
+      // This can literally never happen, and lets us avoid having to catch
+      // UnsupportedEncodingException in callers.
+      throw new AssertionError("UTF-8 is unknown");
```

Comment:
This worries me a bit. The previous version explicitly raised an Stripe exception that you could catch, and now you get an assert that crashes your application right? I don't really grasp why this is a win. You say it can never happen but we do regularly get devs with encoding errors, what if one of their plan or coupon id has an invalid character in their database and such?

Reply:
The `URLEncoder.encode/decode` methods throw an `UnsupportedEncodingException` if you pass an unknown encoding's name. The actual string you're trying to encode is not relevant here. The only case in which this could happen is if you pass an invalid name, and `"UTF-8"` is _always_ a valid name, in all Java environments (it's part of the [specification](https://docs.oracle.com/javase/6/docs/technotes/guides/intl/encoding.doc.html)).

In more recent Java versions, you can do this instead:

```java
String encoded = URLEncoder.encode(stringToEncode, StandardCharsets.UTF_8);
```

and this doesn't throw `UnsupportedEncodingException` because you're directly passing an encoding (not an encoding's name).

So yes, this assert would crash your app if it ever failed, but it will never fail.

Reply:
Ahhhhh yeah okay I see. Damn that was confusing. Reading that code I thought it would throw the same way if the string you passed is in the wrong encoding compared to what you expect. But I see StackOverflow agrees with you so all good!